### PR TITLE
adding google api and google serp api

### PR DESCRIPTION
**Pull request by foreztgump**
first, i want to say thank you so much. this is an amazing project!

i added google api and google serp option to front and back end along with .env and updated docker-compose file

### CODE REVIEW

**Comment by assafelovic** [comment]
Hey @foreztgump thank you for this, great addition! Can you please pull latest changes and adjust? We've merged a PR that actually adds the Tavily search API. We'd love to add Serp + Google as well. Let us know if you have any questions!

Related PR: https://github.com/assafelovic/gpt-researcher/pull/208

**Comment by foreztgump** [comment]
re-adjusted accordingly and also adjust missing api key handling so that the error is more visible in the logs
Screenshot from 2023-10-18 20-19-08

**Comment by foreztgump** [comment]
if you don't mind, I would like to add pdf scraping capability to web_scrape, specifically if arxiv.org/ links are in the list to scrape.

**Comment by assafelovic** [comment]
@foreztgump pdf scraping would be amazing. Lmk when ready to review

**Comment by foreztgump** [comment]
Ready for review. If there anything that you need me to add or change, please let me know. I saw permchain on there. Are you moving toward perrmchain?

File: actions/web_scrape.py [hunk 1]
```diff
@@ -49,23 +51,34 @@ async def async_browse(url: str, question: str, websocket: WebSocket) -> str:
 
     print(f"Scraping url {url} with question {question}")
     await websocket.send_json(
-        {"type": "logs", "output": f"🔎 Browsing the {url} for relevant about: {question}..."})
+        {
+            "type": "logs",
+            "output": f"?? Browsing the {url} for relevant about: {question}...",
```
Comment (assafelovic) [inline actions/web_scrape.py:56]: @foreztgump please modify to original emoji here

File: actions/web_scrape.py [hunk 2]
```diff
@@ -49,23 +51,34 @@ async def async_browse(url: str, question: str, websocket: WebSocket) -> str:
...
-        summary_text = await loop.run_in_executor(executor, summary.summarize_text, url, text, question, driver)
+        summary_text = await loop.run_in_executor(
+            executor, summary.summarize_text, url, text, question, driver
+        )
 
         await websocket.send_json(
-            {"type": "logs", "output": f"📝 Information gathered from url {url}: {summary_text}"})
+            {
+                "type": "logs",
+                "output": f"?? Information gathered from url {url}: {summary_text}",
```
Comment (assafelovic) [inline actions/web_scrape.py:72]: same here

Comment (assafelovic) [reply actions/web_scrape.py:72]: @foreztgump

**Comment by assafelovic** [comment]
@foreztgump added a small comment, looking very good!

**Comment by assafelovic** [comment]
@foreztgump to your question about permchain, we're planning to add several examples of how to leverage GPT Researcher with autonomous frameworks. Next will be autogen. If you're interested in leading it let us know! Love your contribution, thank you! Also please see recent pull, we've just merged another search engine as an option, so please merge it here as well.

File: config/config.py [hunk 3]
```diff
@@ -71,32 +70,65 @@ def set_debug_mode(self, value: bool) -> None:
...
+            )
+
+        return (
+            Fore.RED
+            + f"Please set your {self.service_name} API key in .env or as an environment variable '{service_env}'.\n"
+            + f"You can get your key from {link} \n"
+            + "Alternatively, you can change the 'search_api' value in config.py to 'duckduckgo'."
+        )
 
 def check_config_setup() -> None:
```
Comment (assafelovic) [inline config/config.py:105]: @foreztgump wouldn't it be better to check once depending on config selected search engine instead of calling all?

**Comment by foreztgump** [comment]
all fixed. On your last comment, yes it would be better. I made the adjustment on that as well as the emoji. I'm not sure how it got changed there. Autogen with this would be awesome!! and I wish we have 72 hours a day. I will do my best to contribute.

**Review by assafelovic — APPROVED** [review]

**Comment by assafelovic** [comment]
Thank you @foreztgump for all your great contribution, looking forward to more! Let us know if you'd like to take a stab at autogen (if so open a starting PR), if not we'd be happy to open one. Looking forward!

**Pull request merged**